Client/FileManagerExtension.swift

```diff
@@ -25,4 +27,18 @@ public extension FileManager {
         }
         return true
     }
+    
+    // Check the locked status of a folder. Returns true for locked.
+    public func checkLockedStatus(folder: Folder) -> Bool {
+        let baseDir = NSSearchPathForDirectoriesInDomains(.libraryDirectory, .userDomainMask, true)[0]
+        do {
+            if let lockValue = try self.attributesOfItem(atPath: baseDir + folder.rawValue)[.posixPermissions] as? NSNumber {
+                return lockValue == NSNumber(value: 0o755 as Int16)
+            }
+        } catch let e {
```

Comment:
No need for `let e`, you can just use `error` property inside of the catch closure

Reply:
updated